woocommerce/woocommerce
# [WC_Tracker] Support for get_tracking_history

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

THis PR adds new data into the Tracker Snapshots:

- woocommerce_allow_tracking ( 'yes' or 'no' ) - If woocommerce_allow_tracking is enabled.
- woocommerce_allow_tracking_last_modified (timestamp / 'unknown' ) - Last time the option was changed.
- woocommerce_allow_tracking_first_optin (timestamp  / 'unknown' ) - First time the option was set as 'yes".

PT: pffQ75-1bu-p2



### How to test the changes in this Pull Request:

1. Checkout this PR
2. Run `pnpm -- wp-env run cli -- wp wc tracker snapshot --format=yaml`
3. See the new values at the end of the data stream
4. The expected values should be:
```
  woocommerce_allow_tracking: 'no' 
  woocommerce_allow_tracking_last_modified: "unknown"
  woocommerce_allow_tracking_first_optin: "unknown"
```
5. Run `pnpm -- wp-env run cli -- wp option update woocommerce_allow_tracking 'yes'``
6. Run `pnpm -- wp-env run cli -- wp wc tracker snapshot --format=yaml`
7. Lets assume 123455678 is the current timestamp. Then the expected values should be :
```
  woocommerce_allow_tracking: 'yes' 
  woocommerce_allow_tracking_last_modified: 123455678
  woocommerce_allow_tracking_first_optin:  123455678
```
8. Run `pnpm -- wp-env run cli -- wp option update woocommerce_allow_tracking 'no'`
9.  Run `pnpm -- wp-env run cli -- wp wc tracker snapshot --format=yaml`
10. Lets assume 123123123 is the current timestamp. Then the expected values should be :
```
  woocommerce_allow_tracking: 'no' 
  woocommerce_allow_tracking_last_modified: 123123123
  woocommerce_allow_tracking_first_optin:  123455678
```
11. Run `pnpm -- wp-env run cli -- wp option update woocommerce_allow_tracking 'yes'`
12.  Run `pnpm -- wp-env run cli -- wp wc tracker snapshot --format=yaml`
13. Lets assume 87654321 is the current timestamp. Then the expected values should be :
```
  woocommerce_allow_tracking: 'yes' 
  woocommerce_allow_tracking_last_modified: 87654321
  woocommerce_allow_tracking_first_optin:  123455678
```
14. Clear the options and test again using UI.  (Setting the Opt-in in WooCommerce Onboarding and Changing the values in WooCommerce -> Settings -> Advanced -> WooCommerce.com -> "Allow usage of WooCommerce to be tracked")




### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
